datafusion/physical-plan/src/lib.rs

```diff
@@ -74,7 +74,11 @@ pub trait ExecutionPlan: Debug + DisplayAs + Send + Sync {
     /// If the plan does not support pipelining, but its input(s) are
     /// infinite, returns an error to indicate this.
     fn unbounded_output(&self, _children: &[bool]) -> Result<bool> {
-        Ok(false)
+        if _children.iter().any(|&x| x) {
+            plan_err!("Plan does not support infinite stream from its children")
```

Comment:
While I was surprised initially that this function would return an error, this change seems to make the code match the API described in the comments so 👍

Reply:
If it's alright with you, I'll go ahead and merge it. Thank you for reviewing, @alamb